rbx_xml/src/serializer_core.rs

```diff
@@ -63,6 +63,25 @@ impl<W: Write> XmlEventWriter<W> {
         Ok(())
     }
 
+    /// Writes a `f32` value, but using Roblox's INF, -INF, and NAN values
+    #[inline]
+    pub fn write_characters_f32(&mut self, value: f32) -> Result<(), NewEncodeError> {
+        self.write_characters_f64(value as f64)
+    }
```

Comment:
I'm kind of nervous about casting NaN between `f32` and `f64`. Can we duplicate the logic using the associated consts for `f32`?

Reply:
Casting f32 to f64 is lossless, so nothing should go wrong.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=8775b4db9e81f01f6315154bd72f6e17